repositories/apiserver.go

```diff
@@ -404,6 +408,325 @@ func (a *APIServerStore) StoreCVESummary(ctx context.Context, cve domain.CVEMani
 	return nil
 }
 
+func (a *APIServerStore) StoreVEX(ctx context.Context, cve domain.CVEManifest, cvep domain.CVEManifest, withRelevancy bool) error {
+	_, span := otel.Tracer("").Start(ctx, "APIServerStore.StoreVEX")
+	defer span.End()
+
+	if cve.Name == "" {
+		logger.L().Debug("skipping storing VEX with empty name")
+		return nil
+	}
+
+	// Check if VEX already exists
+	// If it does, update it
+	// If it doesn't, create it
+	vexContainer, err := a.StorageClient.OpenVulnerabilityExchangeContainers(a.Namespace).Get(context.Background(), cve.Name, metav1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Create VEX
+			err = a.createVEX(ctx, cve, cvep, withRelevancy)
+			if err != nil {
+				return err
+			}
+		} else {
+			return err
+		}
+	} else {
+		// Update VEX
+		err = a.updateVEX(ctx, cve, cvep, withRelevancy, vexContainer)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+
+}
+
+func createProductStructForImageAndPackage(imagePullable string, packagePURL string) (*v1beta1.Product, error) {
+	imagePullable = strings.TrimPrefix(imagePullable, "docker://")
+	imageComponents := strings.Split(imagePullable, "/")
+	imageName := imageComponents[len(imageComponents)-1]
+	imageRepo := strings.Join(imageComponents[:len(imageComponents)-1], "/")
+	// pkg:oci/adservice@sha256%3A45fb8ed886902c0c49e044b1f8870fad61c1022fa23c4943098302a8f1c5b75f?repository_url=gcr.io/google-samples/microservices-demo
+	imageField := fmt.Sprintf("pkg:oci/%s?repository_url=%s", url.PathEscape(imageName), url.PathEscape(imageRepo))
+	product := v1beta1.Product{
+		Component: v1beta1.Component{
+			ID: imageField,
+		},
+	}
+	product.Subcomponents = append(product.Subcomponents, v1beta1.Subcomponent{
+		Component: v1beta1.Component{
+			ID: packagePURL,
+		},
+	})
+	return &product, nil
+}
+
+func (a *APIServerStore) createVEX(ctx context.Context, cve domain.CVEManifest, cvep domain.CVEManifest, withRelevancy bool) error {
+	_, span := otel.Tracer("").Start(ctx, "APIServerStore.createVEX")
+	defer span.End()
+
+	imagePullable := cve.Annotations[v1.ImageIDMetadataKey]
+
+	// Timestamp
+	timestamp := time.Now().Format(time.RFC3339)
+
+	// Calculate VEX
+	vexDoc := v1beta1.VEX{
+		Metadata: v1beta1.Metadata{
+			Context:     "https://openvex.dev/ns/v0.2.0",
+			Author:      "kubescape.io",
+			AuthorRole:  "smart vulnerability scanner :-)",
+			Timestamp:   timestamp,
+			LastUpdated: timestamp,
+			Version:     0,
+			Tooling:     "kubescape-vulnerability-analyzer",
```

Comment:
I think we should not have a new tool name here, we should either call this `kubescape`/`kubescape-operator`/`'kubevuln` so it is clear what is the tool.
WDYT?

Reply:
Kubescape and vulnerability is included already. This is ok to release since this is not a production grade feature anyway